src/bfloat16.jl

```diff
@@ -424,3 +432,10 @@ for F in (:abs, :abs2, :sqrt, :cbrt,
      Base.$F(x::BFloat16) = BFloat16($F(Float32(x)))
   end
 end
+
+Base.atan(y::BFloat16, x::BFloat16) = BFloat16(atan(Float32(y), Float32(x)))
+Base.hypot(x::BFloat16, y::BFloat16) = BFloat16(hypot(Float32(x), Float32(y)))
+Base.hypot(x::BFloat16, y::BFloat16, z::BFloat16) = BFloat16(hypot(Float32(x), Float32(y), Float32(z)))
+Base.clamp(x::BFloat16, lo::BFloat16, hi::BFloat16) = BFloat16(clamp(Float32(x), Float32(lo), Float32(hi)))
```

Comment:
Definitions like this are not valuable, to the contrary actually. The generic definitions work fine already: https://github.com/JuliaLang/julia/blob/48ddd2dc2859f41fc1a82876cc5654fb31d7b2ba/base/intfuncs.jl#L1269-L1272

```julia
julia> clamp(BFloat16(5), BFloat16(5.5), BFloat16(6.5))
BFloat16(5.5)
```

By eagerly demoting to `FLoat32` you're making it impossible for LLVM to emit instructions that would support operations like this, which is now possible after https://github.com/JuliaLang/julia/pull/55486.

Similar arguments can be made for other operations.

Reply:
Ok, so this pre-dates the LLVM support. Should we clean up this package and perhaps remove the other operations of this nature?

Reply:
yes

Reply:
Closing this PR in that case.